docs/cloud/satellites/self-hosted.md

```diff
@@ -0,0 +1,215 @@
+# Self-Hosted Satellites
+
+Self-hosted satellites _\*Beta\*_ are a good alternative to Earthly Cloud’s [managed satellites](../satellites.md) when CI/CD is required to run in your own environment. Self-hosted satellites provide most of the benefits of Earthly Cloud while ensuring that build-related data never leaves your network.
+
+In addition, Self-Hosted Satellites have the following features:
+* Lightning-fast caching
+* Cloud-enabled control plane
+* Encrypted connections by default using mTLS
+* Ready to run builds with just a single command
```

Comment:
Perhaps we should mention only the benefits specific to self-hosted satellites, otherwise we're repeating content from https://docs.earthly.dev/earthly-cloud/satellites.

Looping in @thtmnisamnstr in case he has thoughts on messaging here.

Reply:
Side note: We should also amend step 2 from the page I linked above to say something along the lines of "if you want to launch a self-hosted satellite, go to this other page". Otherwise all the steps on that page are the same.

Reply:
Replying to this comment: "Perhaps we should mention only the benefits specific to self-hosted satellites, otherwise we're repeating content from https://docs.earthly.dev/earthly-cloud/satellites."

I removed the "In addition to, ..." wording and updated copy a little, and I like it better. I don't think reiterating the benefits here is a bad thing.